pilot/pkg/model/jwks_resolver.go

```diff
@@ -435,3 +435,41 @@ func (r *JwksResolver) refresh() {
 func (r *JwksResolver) Close() {
 	closeChan <- true
 }
+
+func compareJWKSResponse(oldKeyString string, newKeyString string) bool {
+	var oldJWKs map[string]interface{}
+	var newJWKs map[string]interface{}
+	e := json.Unmarshal([]byte(oldKeyString), &oldJWKs)
+	if e != nil {
+		log.Warnf("Previous JWKs public key JSON is not parseable: %s", oldKeyString)
+		return true
+	}
+	e = json.Unmarshal([]byte(newKeyString), &newJWKs)
+	if e != nil {
+		log.Warnf("New JWKs public key JSON is not parseable: %s", newKeyString)
+		return true
+	}
+
+	if oldJWKs == nil || newJWKs == nil {
+		return true
+	}
+
+	oldKeys := make(map[string]map[string]interface{})
+	for _, oldKey := range oldJWKs["keys"].([]interface{}) {
+		oldKeys[oldKey.(map[string]interface{})["kid"].(string)] = oldKey.(map[string]interface{})
+	}
+	for _, newKey := range newJWKs["keys"].([]interface{}) {
```

Comment:
This doesn't handle the case when the newJWKs have less "kid" and will return false in this case?